iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -165,6 +165,11 @@ typedef struct AMQP_TRANSPORT_DEVICE_TWIN_CONTEXT_TAG
     void* transport_ctx;
 } AMQP_TRANSPORT_DEVICE_TWIN_CONTEXT;
 
+typedef struct AMQP_TRANSPORT_GET_TWIN_CONTEXT_TAG
+{
+    IOTHUB_CLIENT_DEVICE_TWIN_CALLBACK on_get_twin_completed_callback;
+    void* user_context;
+} AMQP_TRANSPORT_GET_TWIN_CONTEXT;
```

Comment:
This is the same structure in the amqp_device.c file.  This can cause confusion to what structures are using what.  Should you have a common amqp function that combines all these similar type structures.

Reply:
They are not exactly the same. They have different callback definitions.